lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregateFuture.java

```diff
@@ -0,0 +1,232 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.reef.annotations.Unstable;
+import org.apache.reef.annotations.audience.ClientSide;
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.annotations.audience.Public;
+import org.apache.reef.vortex.common.VortexFutureDelegate;
+import org.apache.reef.vortex.driver.VortexMaster;
+import org.apache.reef.wake.remote.Codec;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+
+/**
+ * The interface between user code and aggregation Tasklets.
+ * Thread safety: This class is not meant to be used in a multi-threaded fashion.
+ * TODO[JIRA REEF-1131]: Create and run tests once functional.
+ */
+@Public
+@ClientSide
+@NotThreadSafe
+@Unstable
+public final class VortexAggregateFuture<TInput, TOutput> implements VortexFutureDelegate {
+  private static final Logger LOG = Logger.getLogger(VortexAggregateFuture.class.getName());
+
+  private final Executor executor;
+  private final VortexMaster vortexMaster;
+  private final Codec<TOutput> aggOutputCodec;
+  private final BlockingQueue<AggregateResult> resultQueue;
+  private final Map<Integer, Pair<TInput, VortexFunction<TInput, TOutput>>> taskletIdFunctionMap;
+  private final FutureCallback<AggregateResult<TInput, TOutput>> callbackHandler;
+
+  @Private
+  public VortexAggregateFuture(final Executor executor,
+                               final VortexMaster vortexMaster,
+                               final Map<Integer, Pair<TInput, VortexFunction<TInput, TOutput>>> taskletIdFunctionMap,
+                               final Codec<TOutput> aggOutputCodec,
+                               final FutureCallback<AggregateResult<TInput, TOutput>> callbackHandler) {
+    this.executor = executor;
+    this.taskletIdFunctionMap = new HashMap<>(taskletIdFunctionMap);
+    this.resultQueue = new ArrayBlockingQueue<>(taskletIdFunctionMap.size());
+    this.aggOutputCodec = aggOutputCodec;
+    this.callbackHandler = callbackHandler;
+
+    // TODO[JIRA REEF-1129]: Call back from VortexMaster.
+    this.vortexMaster = vortexMaster;
```

Comment:
In the `VortexFuture`(which implements `VortexFutureDelegate`) `VortexMaster` is used only when `cancel()` is requested. As we don't support cancellation for the aggregation, I wonder we use `VortexMaster` in this class.
Could you give some hint if you have a plan to use it in `REEF-1129`?